From eda1dae0e8331eee4d364535dc0685efe43a4531 Mon Sep 17 00:00:00 2001
From: Johannes Schmid <jhs@gnome.org>
Date: Tue, 08 Sep 2009 08:31:20 +0000
Subject: Regenerated gdl.defs and updated dependency to gdl 2.27.92
elelay : taken from https://launchpad.net/ubuntu/natty/+source/gnome-python-extras
         original name is 04_gdl_2.28.patch
	 discarded the configure.ac part
---
diff --git a/gdl/gdl.defs b/gdl/gdl.defs
index 5babad1..795154d 100644
--- gdl/gdl.defs
+++ gdl/gdl.defs
@@ -1,21 +1,10 @@
 ;; -*- scheme -*-
 ; object definitions ...
-(ifdef HAVE_GDL_0_7
-
-(define-object ComboButton
-  (in-module "Gdl")
-  (parent "GtkHBox")
-  (c-name "GdlComboButton")
-  (gtype-id "GDL_TYPE_COMBO_BUTTON")
-)
-
-) ; ifdef HAVE_GDL_0_7
-
-(define-object DockBar
+(define-object DockItemGrip
   (in-module "Gdl")
-  (parent "GtkVBox")
-  (c-name "GdlDockBar")
-  (gtype-id "GDL_TYPE_DOCK_BAR")
+  (parent "GtkContainer")
+  (c-name "GdlDockItemGrip")
+  (gtype-id "GDL_TYPE_DOCK_ITEM_GRIP")
 )
 
 (define-object DockLayout
@@ -46,13 +35,6 @@
   (gtype-id "GDL_TYPE_DOCK_ITEM")
 )
 
-(define-object DockNotebook
-  (in-module "Gdl")
-  (parent "GdlDockItem")
-  (c-name "GdlDockNotebook")
-  (gtype-id "GDL_TYPE_DOCK_NOTEBOOK")
-)
-
 (define-object Dock
   (in-module "Gdl")
   (parent "GdlDockObject")
@@ -60,13 +42,6 @@
   (gtype-id "GDL_TYPE_DOCK")
 )
 
-(define-object DockPaned
-  (in-module "Gdl")
-  (parent "GdlDockItem")
-  (c-name "GdlDockPaned")
-  (gtype-id "GDL_TYPE_DOCK_PANED")
-)
-
 (define-object DockPlaceholder
   (in-module "Gdl")
   (parent "GdlDockObject")
@@ -74,28 +49,8 @@
   (gtype-id "GDL_TYPE_DOCK_PLACEHOLDER")
 )
 
-(define-object DockTablabel
-  (in-module "Gdl")
-  (parent "GtkBin")
-  (c-name "GdlDockTablabel")
-  (gtype-id "GDL_TYPE_DOCK_TABLABEL")
-)
-
-(ifdef HAVE_GDL_0_7
-
-(define-object Icons
-  (in-module "Gdl")
-  (parent "GObject")
-  (c-name "GdlIcons")
-  (gtype-id "GDL_TYPE_ICONS")
-)
-
-) ; ifdef HAVE_GDL_0_7
-
 ;; Enumerations and flags ...
 
-(ifdef HAVE_GDL_0_7
-
 (define-enum DockBarStyle
   (in-module "Gdl")
   (c-name "GdlDockBarStyle")
@@ -108,8 +63,6 @@
   )
 )
 
-) ; ifdef HAVE_GDL_0_7
-
 (define-flags DockItemBehavior
   (in-module "Gdl")
   (c-name "GdlDockItemBehavior")
@@ -127,6 +80,7 @@
     '("cant-dock-center" "GDL_DOCK_ITEM_BEH_CANT_DOCK_CENTER")
     '("cant-close" "GDL_DOCK_ITEM_BEH_CANT_CLOSE")
     '("cant-iconify" "GDL_DOCK_ITEM_BEH_CANT_ICONIFY")
+    '("no-grip" "GDL_DOCK_ITEM_BEH_NO_GRIP")
   )
 )
 
@@ -142,6 +96,20 @@
   )
 )
 
+(define-enum SwitcherStyle
+  (in-module "Gdl")
+  (c-name "GdlSwitcherStyle")
+  (gtype-id "GDL_TYPE_SWITCHER_STYLE")
+  (values
+    '("text" "GDL_SWITCHER_STYLE_TEXT")
+    '("icon" "GDL_SWITCHER_STYLE_ICON")
+    '("both" "GDL_SWITCHER_STYLE_BOTH")
+    '("toolbar" "GDL_SWITCHER_STYLE_TOOLBAR")
+    '("tabs" "GDL_SWITCHER_STYLE_TABS")
+    '("none" "GDL_SWITCHER_STYLE_NONE")
+  )
+)
+
 (define-flags DockParamFlags
   (in-module "Gdl")
   (c-name "GdlDockParamFlags")
@@ -179,49 +147,6 @@
   )
 )
 
-;; From gdl-combo-button.h
-
-(ifdef HAVE_GDL_0_7
-
-(define-function gdl_combo_button_get_type
-  (c-name "gdl_combo_button_get_type")
-  (return-type "GType")
-)
-
-(define-function gdl_combo_button_new
-  (c-name "gdl_combo_button_new")
-  (is-constructor-of "GdlComboButton")
-  (return-type "GtkWidget*")
-)
-
-(define-method set_icon
-  (of-object "GdlComboButton")
-  (c-name "gdl_combo_button_set_icon")
-  (return-type "none")
-  (parameters
-    '("GdkPixbuf*" "pixbuf")
-  )
-)
-
-(define-method set_label
-  (of-object "GdlComboButton")
-  (c-name "gdl_combo_button_set_label")
-  (return-type "none")
-  (parameters
-    '("const-gchar*" "label")
-  )
-)
-
-(define-method set_menu
-  (of-object "GdlComboButton")
-  (c-name "gdl_combo_button_set_menu")
-  (return-type "none")
-  (parameters
-    '("GtkMenu*" "menu")
-  )
-)
-
-) ; ifdef HAVE_GDL_0_7
 
 ;; From gdl-dock-bar.h
 
@@ -230,17 +155,12 @@
   (return-type "GType")
 )
 
-(define-function gdl_dock_bar_new
-  (is-constructor-of "GdlDockBar")
+(define-method bar_new
+  (of-object "GdlDock")
   (c-name "gdl_dock_bar_new")
   (return-type "GtkWidget*")
-  (parameters
-    '("GdlDock*" "dock")
-  )
 )
 
-(ifdef HAVE_GDL_0_7
-
 (define-method get_orientation
   (of-object "GdlDockBar")
   (c-name "gdl_dock_bar_get_orientation")
@@ -271,7 +191,7 @@
   (return-type "GdlDockBarStyle")
 )
 
-) ; ifdef HAVE_GDL_0_7
+
 
 ;; From gdl-dock.h
 
@@ -279,7 +199,6 @@
   (c-name "gdl_dock_new")
   (is-constructor-of "GdlDock")
   (return-type "GtkWidget*")
-  (caller-owns-return #f)
 )
 
 (define-method new_from
@@ -358,10 +277,49 @@
   )
 )
 
+
+
+;; From gdl-dock-item-grip.h
+
+(define-function gdl_dock_item_grip_get_type
+  (c-name "gdl_dock_item_grip_get_type")
+  (return-type "GType")
+)
+
+(define-method grip_new
+  (of-object "GdlDockItem")
+  (c-name "gdl_dock_item_grip_new")
+  (return-type "GtkWidget*")
+)
+
+(define-method set_label
+  (of-object "GdlDockItemGrip")
+  (c-name "gdl_dock_item_grip_set_label")
+  (return-type "none")
+  (parameters
+    '("GtkWidget*" "label")
+  )
+)
+
+(define-method hide_handle
+  (of-object "GdlDockItemGrip")
+  (c-name "gdl_dock_item_grip_hide_handle")
+  (return-type "none")
+)
+
+(define-method show_handle
+  (of-object "GdlDockItemGrip")
+  (c-name "gdl_dock_item_grip_show_handle")
+  (return-type "none")
+)
+
+
+
 ;; From gdl-dock-item.h
 
 (define-function gdl_dock_item_new
   (c-name "gdl_dock_item_new")
+  (is-constructor-of "GdlDockItem")
   (return-type "GtkWidget*")
   (parameters
     '("const-gchar*" "name")
@@ -371,14 +329,13 @@
 )
 
 (define-function gdl_dock_item_new_with_stock
-  (is-constructor-of "GdlDockItem")
   (c-name "gdl_dock_item_new_with_stock")
   (return-type "GtkWidget*")
-  (properties
-    '("name")
-    '("long_name")
-    '("stock_id" (optional))
-    '("behavior" (optional))
+  (parameters
+    '("const-gchar*" "name")
+    '("const-gchar*" "long_name")
+    '("const-gchar*" "stock_id")
+    '("GdlDockItemBehavior" "behavior")
   )
 )
 
@@ -392,9 +349,9 @@
   (c-name "gdl_dock_item_dock_to")
   (return-type "none")
   (parameters
-    '("GdlDockItem*" "target" (null-ok))
+    '("GdlDockItem*" "target")
     '("GdlDockPlacement" "position")
-    '("gint" "docking_param" (default "-1"))
+    '("gint" "docking_param")
   )
 )
 
@@ -422,6 +379,12 @@
   )
 )
 
+(define-method get_grip
+  (of-object "GdlDockItem")
+  (c-name "gdl_dock_item_get_grip")
+  (return-type "GtkWidget*")
+)
+
 (define-method hide_grip
   (of-object "GdlDockItem")
   (c-name "gdl_dock_item_hide_grip")
@@ -506,13 +469,10 @@
   (return-type "GType")
 )
 
-(define-function gdl_dock_layout_new
-  (is-constructor-of "GdlDockLayout")
+(define-method layout_new
+  (of-object "GdlDock")
   (c-name "gdl_dock_layout_new")
   (return-type "GdlDockLayout*")
-  (parameters
-    '("GdlDock*" "dock")
-  )
 )
 
 (define-method attach
@@ -682,21 +642,6 @@
 
 
 
-;; From gdl-dock-notebook.h
-
-(define-function gdl_dock_notebook_new
-  (c-name "gdl_dock_notebook_new")
-  (is-constructor-of "GdlDockNotebook")
-  (return-type "GtkWidget*")
-)
-
-(define-function gdl_dock_notebook_get_type
-  (c-name "gdl_dock_notebook_get_type")
-  (return-type "GType")
-)
-
-
-
 ;; From gdl-dock-object.h
 
 (define-function gdl_dock_object_get_type
@@ -848,24 +793,6 @@
 
 
 
-;; From gdl-dock-paned.h
-
-(define-function gdl_dock_paned_get_type
-  (c-name "gdl_dock_paned_get_type")
-  (return-type "GType")
-)
-
-(define-function gdl_dock_paned_new
-  (c-name "gdl_dock_paned_new")
-  (is-constructor-of "GdlDockPaned")
-  (return-type "GtkWidget*")
-  (properties
-    '("orientation")
-  )
-)
-
-
-
 ;; From gdl-dock-placeholder.h
 
 (define-function gdl_dock_placeholder_get_type
@@ -878,7 +805,7 @@
   (is-constructor-of "GdlDockPlaceholder")
   (return-type "GtkWidget*")
   (parameters
-    '("gchar*" "name")
+    '("const-gchar*" "name")
     '("GdlDockObject*" "object")
     '("GdlDockPlacement" "position")
     '("gboolean" "sticky")
@@ -896,74 +823,49 @@
 
 
 
-;; From gdl-dock-tablabel.h
+;; From gdl.h
+
+
+
+;; From gdl-tools.h
+
 
-(define-function gdl_dock_tablabel_new
-  (c-name "gdl_dock_tablabel_new")
-  (is-constructor-of "GdlDockTablabel")
-  (return-type "GtkWidget*")
-  (properties
-    '("item")
-  )
-)
 
-(define-function gdl_dock_tablabel_get_type
-  (c-name "gdl_dock_tablabel_get_type")
+;; From libgdltypebuiltins.h
+
+(define-function gdl_dock_param_flags_get_type
+  (c-name "gdl_dock_param_flags_get_type")
   (return-type "GType")
 )
 
-(define-method activate
-  (of-object "GdlDockTablabel")
-  (c-name "gdl_dock_tablabel_activate")
-  (return-type "none")
+(define-function gdl_dock_object_flags_get_type
+  (c-name "gdl_dock_object_flags_get_type")
+  (return-type "GType")
 )
 
-(define-method deactivate
-  (of-object "GdlDockTablabel")
-  (c-name "gdl_dock_tablabel_deactivate")
-  (return-type "none")
+(define-function gdl_dock_placement_get_type
+  (c-name "gdl_dock_placement_get_type")
+  (return-type "GType")
 )
 
-;; From gdl-icons.h
-
-(ifdef HAVE_GDL_0_7
-
-(define-function gdl_icons_new
-  (is-constructor-of "GdlIcons")
-  (c-name "gdl_icons_new")
-  (return-type "GdlIcons*")
-  (parameters
-    '("int" "icon_size")
-  )
+(define-function gdl_switcher_style_get_type
+  (c-name "gdl_switcher_style_get_type")
+  (return-type "GType")
 )
 
-(define-function gdl_icons_get_type
-  (c-name "gdl_icons_get_type")
+(define-function gdl_dock_item_behavior_get_type
+  (c-name "gdl_dock_item_behavior_get_type")
   (return-type "GType")
 )
 
-(define-method get_folder_icon
-  (of-object "GdlIcons")
-  (c-name "gdl_icons_get_folder_icon")
-  (return-type "GdkPixbuf*")
+(define-function gdl_dock_item_flags_get_type
+  (c-name "gdl_dock_item_flags_get_type")
+  (return-type "GType")
 )
 
-(define-method get_uri_icon
-  (of-object "GdlIcons")
-  (c-name "gdl_icons_get_uri_icon")
-  (return-type "GdkPixbuf*")
-  (parameters
-    '("const-char*" "uri")
-  )
+(define-function gdl_dock_bar_style_get_type
+  (c-name "gdl_dock_bar_style_get_type")
+  (return-type "GType")
 )
 
-(define-method get_mime_icon
-  (of-object "GdlIcons")
-  (c-name "gdl_icons_get_mime_icon")
-  (return-type "GdkPixbuf*")
-  (parameters
-    '("const-char*" "mime_type")
-  )
-)
 
-) ; ifdef HAVE_GDL_0_7
--
cgit v0.8.2
